dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleAPIImpl.java

```diff
@@ -82,7 +111,234 @@ public Bundle getBundleById(String id) throws DotDataException {
+		}*/
+		// todo: bundle is not a permissionable yet, so can not validate yet.
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteBundleAndDependenciesOlderThan(final Date olderThan, final User user) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final boolean isAdmin = this.userAPI.isCMSAdmin(user);
+
+		try {
+
+			Logger.info(this, "Deleting bundles older than: " + olderThan);
+			try (final Stream<String> bundleIds = this.getOlderBundleIds(olderThan, user.getUserId(), isAdmin)) {
+
+				bundleIds.forEachOrdered( bundleId -> {
+
+					this.internalDeleteBundleAndDependencies(bundleId, user);
+					bundlesDeleted.add(bundleId);
+				});
+			}
+		} catch (IOException e) {
+
+			throw new DotDataException(e);
+		}
+
+		return bundlesDeleted.build();
+	}
+
+	private Stream<String> getOlderBundleIds(final Date olderThan, final String userId, final boolean isAdmin)
+			throws IOException, DotDataException {
+
+		final File tempFile       = FileUtil.createTemporalFile("bundle-older-delete");
+		final int limit           = 100;
+		int offset                = 0;
+		List<Bundle> sentBundles  = isAdmin?
+				this.bundleFactory.findSentBundles(olderThan, limit, offset):
+				this.bundleFactory.findSentBundles(olderThan, userId, limit, offset);
+		BufferedWriter fileWriter = null;
+
+		// if the initial is less than limit, return it
+		if (UtilMethods.isSet(sentBundles) && sentBundles.size() < limit) {
+
+			return sentBundles.stream().map(Bundle::getId);
+		}
+
+		try {
+
+			fileWriter = new BufferedWriter(new FileWriter(tempFile));
+			while (UtilMethods.isSet(sentBundles)) {
+
+				for (final Bundle bundle : sentBundles) {
+
+					fileWriter.write(bundle.getId());
+					fileWriter.newLine();
+				}
+
+				fileWriter.flush();
+
+				offset       += limit + 1;
+				sentBundles   = isAdmin?
+						this.bundleFactory.findSentBundles(olderThan, limit, offset):
+						this.bundleFactory.findSentBundles(olderThan, userId, limit, offset);
+			}
+		} finally {
+
+			CloseUtils.closeQuietly(fileWriter);
+		}
+
+		return Files.lines(tempFile.toPath());
+	}
+
+	private void internalDeleteBundleAndDependencies(final String bundleId, final User user) {
+
+		try {
+
+			this.deleteBundleAndDependencies(bundleId, user);
+		} catch (DotDataException e) {
+
+			throw new DotRuntimeException(e);
+		}
+	}
+
+	@WrapInTransaction
+	@Override
+	public Set<String> deleteAllBundles(final User user,
+										 final PublishAuditStatus.Status ...statuses) throws DotDataException {
+
+		final ImmutableSet.Builder<String> bundlesDeleted = new ImmutableSet.Builder<>();
+		final int limit          = 100;
+		final int offset         = 0;
+		final boolean isAdmin    = this.userAPI.isCMSAdmin(user);
+		final List<Status> statusList = Arrays.asList(statuses);
+		List<String> bundleIDs = isAdmin ? publishAuditAPI.getBundleIdByStatus(statusList,limit,offset) :
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4502981)